my_package/compiler_args.py

```diff
@@ -44,7 +47,58 @@
     # '-gencode', 'arch=compute_70,code=sm_70',
     # '-gencode', 'arch=compute_70,code=compute_70'
 
-    '-w' # Ignore compiler warnings.
 ]
 
-cxx_args = ['-std=c++11', '-w']
\ No newline at end of file
+CUDA_SUCCESS = 0
+
+def exec_cuda(descr, func):
+    result = func()
+    if result != CUDA_SUCCESS:
+        cuda.cuGetErrorString(result, ctypes.byref(error_str))
+        raise OSError(f"{descr} failed with error code {result}: {error_str.value.decode()}")
+
+try:
+    nvcc_args = []
+
+    # Copy pasted from snippet of Jan Schlüter, https://gist.github.com/edgarsi/ba0554229bff714a900f2766b0715114
+    import ctypes
+    libnames = ('libcuda.so', 'libcuda.dylib', 'cuda.dll')
+    for libname in libnames:
+        try:
+            cuda = ctypes.CDLL(libname)
+        except OSError:
+            continue
+        else:
+            break
+    else:
+        raise OSError("could not load any of: " + ' '.join(libnames))
+
+    error_str = ctypes.c_char_p()
+
+    exec_cuda('cuInit', lambda: cuda.cuInit(0))
+    nGpus = ctypes.c_int()
+    exec_cuda('cuDeviceGetCount', lambda: cuda.cuDeviceGetCount(ctypes.byref(nGpus)))
+    print("Found %d device(s)." % nGpus.value)
+    for i in range(nGpus.value):
+        device = ctypes.c_int()
+        exec_cuda('cuDeviceGet', lambda: cuda.cuDeviceGet(ctypes.byref(device), i))
+        cc_major = ctypes.c_int()
+        cc_minor = ctypes.c_int()
+        exec_cuda('cuDeviceComputeCapability', lambda: cuda.cuDeviceComputeCapability(ctypes.byref(cc_major), ctypes.byref(cc_minor), device))
+        compat = f'{cc_major.value}{cc_minor.value}'
+        nvcc_args.extend([
+            '-gencode', f'arch=compute_{compat},code=sm_{compat}',
+        ])
+        
+except Exception as e:
+    print(e)
+    nvcc_args = safe_nvcc_args

```

Comment:
I love how you auto-detected the compatibility needed and yet fell back on our existing approach if that didn't work. Thank you very much for doing this kind of backwards-compatibility!